pkg/controllers/osc/osc_controller.go

```diff
@@ -277,3 +228,18 @@ func (r *Reconciler) deleteGeneratedSecrets(ctx context.Context, md *clusterv1al
 	}
 	return nil
 }
+
+func (r *Reconciler) SetupWithManager(mgr manager.Manager) error {
```

Comment:
I am not quite sure about the difference between this, and explicitly call the watch as every controller. I guess we can adopt that, however can you please revert the old Add function and include this in it.